web/src/utils/formatting.test.ts

```diff
@@ -247,3 +256,110 @@ describe('formatCo2', () => {
     expect(actual).to.eq(expected);
   });
 });
+
+describe('getDateTimeFormatOptions', () => {
+  it('handles hourly data', () => {
+    const actual = getDateTimeFormatOptions(TimeAverages.HOURLY);
+    const expected = {
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric',
+      hour: 'numeric',
+      minute: 'numeric',
+      timeZoneName: 'short',
+    };
+    expect(actual).to.deep.eq(expected);
+  });
+  it('handles daily data', () => {
+    const actual = getDateTimeFormatOptions(TimeAverages.DAILY);
+    const expected = {
+      dateStyle: 'long',
+      timeZone: 'UTC',
+    };
+    expect(actual).to.deep.eq(expected);
+  });
+  it('handles monthly data', () => {
+    const actual = getDateTimeFormatOptions(TimeAverages.MONTHLY);
+    const expected = {
+      month: 'long',
+      year: 'numeric',
+      timeZone: 'UTC',
+    };
+    expect(actual).to.deep.eq(expected);
+  });
+  it('handles yearly data', () => {
+    const actual = getDateTimeFormatOptions(TimeAverages.YEARLY);
+    const expected = {
+      year: 'numeric',
+      timeZone: 'UTC',
+    };
+    expect(actual).to.deep.eq(expected);
+  });
+  it('logs an error on unknown data', () => {
+    // Spy on console.error to check if it is called
+    const consoleErrorSpy = vi.spyOn(console, 'error');
+
+    const actual = getDateTimeFormatOptions('ThisAggregateDoesNotExist' as TimeAverages);
+    const expected = {};
+    expect(actual).to.deep.eq(expected);
+    expect(consoleErrorSpy).toHaveBeenCalledWith(
+      'ThisAggregateDoesNotExist is not implemented'
+    );
+
+    // Restore the spy
+    consoleErrorSpy.mockRestore();
+  });
+});
+
+describe('formatDate', () => {
+  it('handles hourly data', () => {
+    const actual = formatDate(
+      new Date('2021-01-01T00:00:00Z'),
+      'en',
+      TimeAverages.HOURLY
+    );
+    const expected = 'January 1, 2021 at 1:00 AM GMT+1';
+    expect(actual).to.deep.eq(expected);
+  });
```

Comment:
Can we create one more test on another language than en?

Reply:
Sure thing, I can change this to a snapshot test and then we can add all the languages we want :)